src/main/kotlin/org/rust/cargo/project/model/impl/CargoTomlWatcher.kt

```diff
@@ -57,20 +61,39 @@ class CargoTomlWatcher(
         )
 
         @VisibleForTesting
-        fun isInterestingEvent(event: VFileEvent): Boolean = when {
-            event.pathEndsWith(CargoConstants.MANIFEST_FILE) || event.pathEndsWith(CargoConstants.LOCK_FILE) -> true
-            event is VFileContentChangeEvent -> false
-            !event.pathEndsWith(".rs") -> false
-            event is VFilePropertyChangeEvent && event.propertyName != VirtualFile.PROP_NAME -> false
-            IMPLICIT_TARGET_FILES.any { event.pathEndsWith(it) } -> true
-            else -> {
-                val parent = PathUtil.getParentPath(event.path)
-                val grandParent = PathUtil.getParentPath(parent)
-                IMPLICIT_TARGET_DIRS.any { parent.endsWith(it) || (event.pathEndsWith("main.rs") && grandParent.endsWith(it)) }
+        fun isInterestingEvent(project: Project, event: VFileEvent): Boolean {
+            return when {
+                event.pathEndsWith(CargoConstants.MANIFEST_FILE) -> true
+                event.pathEndsWith(CargoConstants.LOCK_FILE) -> {
+                    val projectDir = Paths.get(event.path).parent
+                    val timestamp = CargoEventService.getInstance(project).extractTimestamp(projectDir) ?: 0
+                    if (event.requestor != null) return true
+                    val current = System.currentTimeMillis()
+                    val delayThreshold = Registry.intValue("org.rust.cargo.lock.update.delay.threshold")
+                    val delay = current - timestamp
+                    return if (delay > delayThreshold) {
+                        LOG.info("External change in ${event.path}. Previous Cargo metadata call was $delay ms before")
+                        true
+                    } else {
+                        LOG.info("Skip external change for ${event.path}. Previous Cargo metadata call was $delay ms before")
```

Comment:
Maybe `debug`?

Reply:
It doesn't look this message will appear in logs too frequently (only on cargo metadata refresh after some change in Cargo.toml or at the first time) so I'd prefer to keep it enabled by default to catch potential problems from logs and not to ask users to enable them